### ISSUE TITLE
Map extras do not evolve monsters

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#68474]
### Describe the bug

Every other method of spawning them respects evolution.

### Attach save file

n/a, see #68458 instead.

### Steps to reproduce

1. Have a map extra spawning e.g civilians like my PR above.
2. Set the time to e.g 5 years.
3. Civilians still spawn in those map extras.

### Expected behavior

They get evolved.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: d48717d [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by ZhilkinSerg** [comment]
I guess map extras should have optional start and end day fields.

**Comment by Procyonae** [comment]
Commented a solution I think works on the PR

**Comment by Procyonae** [comment]
I'd argue that manual placement of unevolved monsters should remain possible however this gets resolved

**Comment by NetSysFire** [comment]
Yeah, just apply evolution by default unless you specify `evolve: false` or something in the entry.

**Comment by Procyonae** [comment]
I'm also not sure this is exclusive to map extras but I haven't tested it

**Comment by I-am-Erk** [comment]
per discussion in discord, I would like to check: is it because of map extra behaviour, or because they are being placed as specific monsters by ID rather than part of a mongroup? Placing specific monsters --> not evolving would be intended behaviour, the solution then if you want a specific civilian would be to make a mongroup with just that civilian in it. Otherwise we're going to get some really annoying behaviour in other places where people have intentionally used the previous behaviour.
